Microsoft.Toolkit.Win32/Microsoft.Toolkit.Wpf.UI.XamlHost/WindowsXamlHostBase.cs

```diff
@@ -0,0 +1,187 @@
+        /// deinitialized when the last instance of WindowsXamlManager is destroyed.
+        /// </summary>
+        private readonly Windows.UI.Xaml.Hosting.WindowsXamlManager _windowsXamlManager;
+
+        /// <summary>
+        ///     Fired when WindowsXamlHost root UWP XAML content has been updated
+        /// </summary>
+        public event EventHandler ChildChanged;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="WindowsXamlHostBase"/> class.
+        /// </summary>
+        /// <remarks>
+        /// Default constructor is required for use in WPF markup. When the default constructor is called,
+        /// object properties have not been set. Put WPF logic in OnInitialized.
+        /// </remarks>
+        public WindowsXamlHostBase()
+        {
+            // Windows.UI.Xaml.Application object is required for loading custom control metadata.  If a custom
+            // Application object is not provided by the application, the host control will create one (XamlApplication).
+            // Instantiation of the application object must occur before creating the DesktopWindowXamlSource instance.
+            // If no Application object is created before DesktopWindowXamlSource is created, DestkopWindowXamlSource
+            // will create a generic Application object unable to load custom UWP XAML metadata.
+            Microsoft.Toolkit.Win32.UI.XamlHost.XamlApplication.GetOrCreateXamlApplicationInstance(ref _application);
+
+            // Create an instance of the WindowsXamlManager. This initializes and holds a
+            // reference on the UWP XAML DXamlCore and must be explicitly created before
+            // any UWP XAML types are programmatically created.  If WindowsXamlManager has
+            // not been created before creating DesktopWindowXamlSource, DesktopWindowXaml source
+            // will create an instance of WindowsXamlManager internally.  (Creation is explicit
+            // here to illustrate how to initialize UWP XAML before initializing the DesktopWindowXamlSource.)
+            _windowsXamlManager = Windows.UI.Xaml.Hosting.WindowsXamlManager.InitializeForCurrentThread();
+
+            // Create DesktopWindowXamlSource, host for UWP XAML content
+            _xamlSource = new Windows.UI.Xaml.Hosting.DesktopWindowXamlSource();
+
+            // Hook OnTakeFocus event for Focus processing
+            _xamlSource.TakeFocusRequested += OnTakeFocusRequested;
+        }
+
+        /// <summary>
+        /// Gets or sets the root UWP XAML element displayed in the WPF control instance.
+        /// </summary>
+        /// <value>The <see cref="Windows.UI.Xaml.UIElement"/> child.</value>
+        /// <remarks>This UWP XAML element is the root element of the wrapped <see cref="Windows.UI.Xaml.Hosting.DesktopWindowXamlSource" />.</remarks>
+        public Windows.UI.Xaml.UIElement ChildInternal
```

Comment:
If this is public why permit `SetContent` to be protected? Usage in sample controls doesn't make sense as none of them use the arguments